ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/TableSelect.java

```diff
@@ -68,9 +71,14 @@ private TableSelect(
       @JsonProperty(value = "source", required = true) final ExecutionStep<KTableHolder<K>> source,
       @JsonProperty(value = "keyColumnNames") final Optional<List<ColumnName>> keyColumnNames,
       @JsonProperty(value = "selectExpressions", required = true) final
-      List<SelectExpression> selectExpressions
+      List<SelectExpression> selectExpressions,
+      @JsonProperty(value = "internalFormats", required = true) final Formats internalFormats
```

Comment:
This will change ~650 historical plans

Reply:
Will KSQL still be able to load historical plans that don't include this field?

Reply:
I would think you would have to remove required = true to handle it without.  I thought the idea was that this should exist only for plans made by newer ksql versions.

Reply:
I'm assuming that this is the internal format of the materialization if it's to be materialized?  Maybe it makes more sense to wrap it in another object called Materialization with format as a field if we think there could be more metadata?  Or just call this materializationFormat?